src/ctfeexpr.d

```diff
@@ -76,12 +76,12 @@ public:
     {
         ClassDeclaration cd = originalClass();
         uint fieldsSoFar = 0;
-        while (index - fieldsSoFar >= cd.fields.dim)
+        while ((value.elements.dim - fieldsSoFar - index) > cd.fields.dim)
         {
             fieldsSoFar += cd.fields.dim;
             cd = cd.baseClass;
         }
-        return cd.fields[index - fieldsSoFar];
+        return cd.fields[cd.fields.dim - (value.elements.dim - fieldsSoFar - index)];
     }
```

Comment:
Looks OK to me, I'd have to try it out for the other two functions also... Maybe :-)

I probably shouldn't, but my eyes become twitchy when I see a calculated index that first underflows then overflows to be back within the array range. ;-)